### DISCUSSION TITLE
Contributions from baseline/intercept, trend and extra features

### DISCUSSION THREAD

**Discussion by xijianlim** [google/lightweight_mmm#28] [General]
Hi, i've noticed that the examples have 0% contribution from the baseline when running the functions:
plot_media_baseline_contribution_area_plot or create_media_baseline_contribution_df in your example notebooks; with media cannels having a 100% contribution.

In addition, i've the code base over a data set of mine and this was also the case. This would mean even the "extra features" contribution default to null or 0 contribution, which makes interpretation a little strange when suggesting media is 100% influencing the KPI target.

Is there a way to set or tune the baseline/intercept so that is has a ball park contribution? Are there any suggestions you would suggest in the mean time?

**Comment by pabloduque0** [comment]
hey @xijianlim  !

As for what the examples concern, it is dummy data and the way it is generated it does have a very small baseline so I would not worry too much for that case. That said we can probably increase the baseline of the dummy data to be reflected by the model as well.

Okay if you run into this situation with actual data we can definitely look into it.

The easiest way to alter any of this is through custom priors (eg. decreasing media priors and increasing other priors such as `intercept` or `extra_features`). Im working on a better API for passing custom priors which should be ready very soon. I can notify you here when its released.

**Reply by xijianlim** [reply]
Thanks got your quick response as always Pablo!

Will your plot/df functions also be able to provide the contributions from the extra features as well?

**Reply by xijianlim** [reply]
I've also used actual data and found it had no baseline contributions in the output of create_media_baseline_contribution_df

**Reply by pabloduque0** [reply]
Can you try reducing your media priors so I can get another input and investigate this?

```
mmm.fit(
    media=...,
    total_costs=costs / 10,
    target=...,
    extra_features=...,
    number_warmup=...,
    number_samples=...,
    number_chains=...,
    seed=...)
```

10 might not be the right divisor but it is just for testing sake

**Reply by pabloduque0** [reply]
At the moment the approach is to have the extra features as non media baseline, but we can definitely look into separating those.

**Reply by xijianlim** [reply]
Hey Pablo, this is the results of dividing cost by 10: 
![image](https://user-images.githubusercontent.com/25444256/176299893-1c2ad3fd-0015-4ec3-ae90-2b1ec0766094.png)

Thanks for the update!

**Reply by saukang** [reply]
Is there any update on separating those extra features from the baseline? That will be a brilliant idea!

**Reply by xijianlim** [reply]
You can modify the function yourself to calculate the decompositon values:

something like this from the mean
`extra_features_contrib=media_mix_model.trace['beta_extra_features'].mean(axis=0)*extra_features`

I took the plot function and made my own modification to produce a dataframe. You just need to then use the trace function to get the intercept as well if needed.

**Reply by saukang** [reply]
Thanks buddy! Yeah I just realised that we can get that from `trace`! Then we can concat it with the media contribution df and plot the whole picture out! Cheers!

**Reply by alexborio** [reply]
> Can you try reducing your media priors so I can get another input and investigate this?
> 
> ```
> mmm.fit(
>     media=...,
>     total_costs=costs / 10,
>     target=...,
>     extra_features=...,
>     number_warmup=...,
>     number_samples=...,
>     number_chains=...,
>     seed=...)
> ```
> 
> 10 might not be the right divisor but it is just for testing sake

What purpose does this scaling achieve and is there a way to find a proper divisor? Cost scaling affects the baseline contribution, but can't grasp the idea of why and when to apply it.

**Reply by pabloduque0** [reply]
You can do it in a more clean way directly when scaling the costs:
```
costs_scaler = preprocessing.CustomScaler(divide_operation=jnp.mean, multiply_by=0.1)
# then apply scaler
```

After all you are setting the scale of your priors, that will depend on your specific case, your data and what priors you want to set. You should have some ballpark idea of what media contribution you should have (same for other metrics/insights),  you can test different scales of your priors that achieves both good metric models and also makes sense based on what you expect it to look like. Or you can also have a more heuristic approach.

**Reply by rafpinter** [reply]
Hi everyone!
I'm trying to get the contribution from the extra features as well.

> You can modify the function yourself to calculate the decompositon values:
> 
> something like this from the mean `extra_features_contrib=media_mix_model.trace['beta_extra_features'].mean(axis=0)*extra_features`
> 
> I took the plot function and made my own modification to produce a dataframe. You just need to then use the trace function to get the intercept as well if needed.

@xijianlim where exactly did you implement this modification? Would you mind providing more details on the code? That would help me a lot.

Thanks!

**Reply by xijianlim** [reply]
Hi i implemented this mod by basically reverse engineering the plot function: "plot_media_baseline_contribution_area_plot". 

You can use the trace function to get the model's multiplier in something like this:


`extra_features_contrib=media_mix_model.trace['extra_features'].mean(axis=0)*extra_features

**Reply by AruunnAnanda** [reply]
Hi @xijianlim, thanks for sharing the code to deduce the contribution by extra features. 

I've two questions - 

1. The column that you actually meant was this -  **mmm.trace['coef_extra_features']**, right? 
2. Shouldn't we inverse transform to get the real contribution from these extra features? The co-efficients of extra features refer to the scaled values, right? 

I'm not a data scientist, but rather an analyst. Sorry for my lame questions. Thanks!!

**Reply by xijianlim** [reply]
sorry ive been away for too long. short answer is **yes**; you should inverse transform these variables.
